pkg/utils/utils.go

```diff
@@ -38,59 +54,77 @@ func RandomString(length int) (string, error) {
 // buildCSPDirectives takes the content security policy configuration from the server and constructs
 // a complete set of directives for the Content-Security-Policy-Report-Only header.
 // The constructed directives will include the default sources and the supplied configuration.
+
 func BuildCSPDirectives(k8sMode, pluginsCSP, indexPageScriptNonce string) ([]string, error) {
+	nonce := fmt.Sprintf("'nonce-%s'", indexPageScriptNonce)
+
 	// The default sources are the sources that are allowed for all directives.
-	// When running on-cluster, the default sources are just 'self' (i.e. the same origin).
-	// When running off-cluster, the default sources are 'self' and 'http://localhost:8080' and 'ws://localhost:8080'
-	// (i.e. the same origin and the proxy endpoint).
-	defaultSources := "'self'"
+	// When running on-cluster, the default sources are just 'self' and 'console.redhat.com'.
+	// When running off-cluster, 'http://localhost:8080' and 'ws://localhost:8080' are appended to the
+	// default sources. Image source, font source, and style source only use 'self' and
+	// 'http://localhost:8080'.
+	baseUriDirective := []string{baseURI, self, consoleDot}
+	defaultSrcDirective := []string{defaultSrc, self, consoleDot}
```

Comment:
It still feels like console dot should be a `connect-src` rather than `default-src`.